CreateMessage/handler.ts

```diff
@@ -307,6 +361,63 @@ export function CreateMessageHandler(
     const { service, email: serviceUserEmail } = userAttributes;
     const { authorizedRecipients, serviceId } = service;
 
+    // Check configuration id if third_party_data has been sent
+    const isRemoteContentMessage = !!messagePayload.content.third_party_data;
+    const hasRemoteContentMessageConfigurationId = !!messagePayload.content
+      .third_party_data?.configuration_id;
+
+    if (isRemoteContentMessage && !hasRemoteContentMessageConfigurationId) {
+      return ResponseErrorForbiddenNoConfigurationId;
+    }
+
+    if (hasRemoteContentMessageConfigurationId) {
+      // Get the remote content configuration and check status codes
+      const getRCConfigurationResponseOrError = await messagesServicesApiClient.getRCConfiguration(
+        {
+          configurationId:
+            messagePayload.content.third_party_data.configuration_id
+        }
+      );
+
+      if (E.isLeft(getRCConfigurationResponseOrError)) {
+        return ResponseErrorInternal(
+          "Cannot retrieve the remote content configuration. Service unreachable."
+        );
+      }
+
+      switch (getRCConfigurationResponseOrError.right.status) {
+        case 400:
+          return ResponseErrorValidation(
+            "Validation error",
+            "The request to get the remote content configuration is invalid."
+          );
+        case 403:
+          return ResponseErrorForbiddenNotAuthorized;
+        case 404:
+          return ResponseErrorNotFound(
+            "Remote Content Configuration NOT FOUND",
+            "Cannot find any Remote Content Configuration with the given id"
+          );
+        case 500:
+          return ResponseErrorInternal(
+            "Generic error happened while trying to get remote content configuration."
+          );
+        case 200:
+          // check the ownership of the remote content configuration if sent with third_party_data
+          if (
+            parseOwnerIdFullPath(auth.userId) !==
+            getRCConfigurationResponseOrError.right.value.user_id
```

Comment:
seems like that there are multiple names that refers to the same thing (`ownerId`, `userId`)

Reply:
where is `ownerId`?